ynput/ayon-hiero
# Fix timecode detection and image sequence source range on OTIO AYON exporter for Hiero.

## Changelog Description
resolve #13 

Fixes custom AYON OTIO export.
* embedded timecode not detected properly for image sequence and movies
* image sequence `source_range` was relative to the `available_range`

## Additional info

* Tested again OTIO result from native **Hiero v15.1v3** (in previous version OTIO export is flagged as Beta).
* **Native Hiero v15.1v3 does not return conformed fps clip per sequence.
E.g. here output sequence is 25fps, Hiero still defines some valid range but they are based on 24fps.
AYON custom export produce conformed ranges.** 
![image](https://github.com/user-attachments/assets/f7c3093f-4ebd-42c7-9fd5-2b3c13b37c95)

`87399.0 / 24.0 == 91040.625 / 25.0`


## Testing notes:

Ensure that the `available_range` and `source_range` match for each clip with exporting OTIO from native Hiero and custom AYON.

1. export with AYON custom exporter
```python
from ayon_hiero.api.otio import hiero_export

output_path = r"C:\path\to\output\custom.otio"
otio_timeline = hiero_export.create_otio_timeline()        
hiero_export.write_to_file(otio_timeline, output_path)
```

2. export native Hiero exporter
```python
import hiero.core

preset_name = "OTIO"
preset = hiero.core.taskRegistry.processorPresetByName(preset_name)
item = hiero.core.ItemWrapper(sequence_obj)
hiero.core.taskRegistry.createAndExecuteProcessor(preset, [item])
```

Here are my results (made a timeline with a mix of timeline:
* consistent fps across timeline and source media =  [timeline_fps_consistent.zip](https://github.com/user-attachments/files/17495845/timeline_fps_consistent.zip)
* different fps between timeline and source media =  [timeline_fps_mismatch.zip](https://github.com/user-attachments/files/17495846/timeline_fps_mismatch.zip)
